src/main/java/jmgomez/apipolicy/mapper/MapStructMapper.java

```diff
@@ -0,0 +1,23 @@
+package jmgomez.apipolicy.mapper;
+
+import jmgomez.apipolicy.model.Accident;
+import jmgomez.apipolicy.model.Policy;
+import jmgomez.apipolicy.model.dto.AccidentDto;
+import jmgomez.apipolicy.model.dto.PolicyDto;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface MapStructMapper {
```

Comment:
A Mapper should be created for each model and you are missing the opportunity to manage here the collections and avoid the complicated code you have in the Controller

Reply:
Also, you can use `@Mapper(componentModel = "spring")` and avoid entirely the `jmgomez.apipolicy.config.MapStructConfig.class` so you would not need to define every bean for every mapper.